packages/notification-center/src/store/styles/styles-provider.types.ts

```diff
@@ -0,0 +1,111 @@
+import { ObjectWithRoot, CSSFunctionOrObject } from '../../components/types';
+
+export type StylesPaths =
+  | 'bellButton.root'
+  | 'bellButton.dot'
+  | 'popover.arrow'
+  | 'popover.dropdown'
+  | 'loader.root'
+  | 'layout.root'
+  | 'header.root'
+  | 'header.title'
+  | 'header.markAsRead'
+  | 'header.cog'
+  | 'header.backButton'
+  | 'tabs.tabsList'
+  | 'tabs.tab'
+  | 'tabs.tabLabel'
+  | 'tabs.tabIcon'
+  | 'accordion.item'
+  | 'accordion.content'
+  | 'accordion.control'
+  | 'accordion.chevron'
+  | 'switch.root'
+  | 'switch.input'
+  | 'switch.track'
+  | 'switch.thumb'
+  | 'unseenBadge.root'
+  | 'footer.root'
+  | 'footer.title'
+  | 'notifications.root'
+  | 'notifications.listItem.read'
+  | 'notifications.listItem.unread'
+  | 'notifications.listItem.layout'
+  | 'notifications.listItem.contentLayout'
+  | 'notifications.listItem.title'
+  | 'notifications.listItem.timestamp'
+  | 'notifications.listItem.buttons.root'
+  | 'notifications.listItem.buttons.primary'
+  | 'notifications.listItem.buttons.secondary'
+  | 'preferences.root'
+  | 'preferences.item.title'
+  | 'preferences.item.channels'
+  | 'preferences.item.divider'
+  | 'preferences.item.content.icon'
+  | 'preferences.item.content.channelLabel'
+  | 'preferences.item.content.success';
```

Comment:
I didn't know how to do this better, I've tried to write the type that will return from the object all leaves, but the thing is that each prop has a different level of deepness and the styles objects don't have a limit for its props... so I've failed here and decided to define the paths like this, it's not ideal but the pros is that when someone will use the `useStyles` hook then the TS will show all possible paths... these strings should correspond to the leaves in the below object

Reply:
This solution works well for our case, so it is good enough. Not need for _better_. 🙂